### DISCUSSION TITLE
How to auto-materialize historical partitions of a time-partitioned asset?

### DISCUSSION THREAD

**Discussion by sryza** [dagster-io/dagster#15567] [Q&A] [answered]
How to make sure that if a number of latest time partitioned asset partitions are missing, all of it are triggered for materialization? Currently if I understand correctly only the latest is triggered, if auto materialization is used. Any other approaches that would guarantee no missing non-materialized partitions?

**Comment by sryza** [comment, marked as answer]
The `max_materializations_per_minute` of `AutoMaterializePolicy` is what governs this.  Its default value is `1`. This means that, if a bunch of historical partitions would be auto-materialized at once, all partitions except for the latest one will be discarded.

The goal of this parameter is to avoid "surprise backfills", where an accidental change to upstream data or similar causes a large number of partitions to be auto-materialized.

You can change this behavior by changing the `max_materializations_per_minute` value. If you set it to `None`, there will be no restrictions on the number of historical partitions materialized.

**Reply by derHeinzer** [reply]
@sryza is there a way to have the backfill policy respected? In my case I dont want max_materializations_per_minute single backfill runs, each auto-materializiing a single partition. I would need multiple runs with max_partitions_per_run = 100 e.g.

Any piece of advise is highly appreciated :)

**Reply by sryza** [reply]
This isn't currently possible, but I filed an issue to track the request: https://github.com/dagster-io/dagster/issues/18237
